src/Telemetry/AzureDevOpsTimeline/KustoTimelineTelemetryRepository.cs

```diff
@@ -0,0 +1,197 @@
+                    new KustoValue("Path", b.AugmentedOrder, KustoDataType.String),
+                    new KustoValue("ParentId", b.Raw.ParentId, KustoDataType.String),
+                    new KustoValue("Name", b.Raw.Name, KustoDataType.String),
+                    new KustoValue("StartTime", b.Raw.StartTime, KustoDataType.DateTime),
+                    new KustoValue("FinishTime", b.Raw.FinishTime, KustoDataType.DateTime),
+                    new KustoValue("Result", b.Raw.Result, KustoDataType.String),
+                    new KustoValue("ResultCode", b.Raw.ResultCode, KustoDataType.String),
+                    new KustoValue("ChangeId", b.Raw.ChangeId, KustoDataType.Int),
+                    new KustoValue("LastModified", b.Raw.LastModified, KustoDataType.DateTime),
+                    new KustoValue("WorkerName", b.Raw.WorkerName, KustoDataType.String),
+                    new KustoValue("Details", b.Raw.Details?.Url, KustoDataType.String),
+                    new KustoValue("ErrorCount", b.Raw.ErrorCount, KustoDataType.Int),
+                    new KustoValue("WarningCount", b.Raw.WarningCount, KustoDataType.Int),
+                    new KustoValue("Url", b.Raw.Url, KustoDataType.String),
+                    new KustoValue("LogId", b.Raw.Log?.Id, KustoDataType.Int),
+                    new KustoValue("LogUri", b.Raw.Log?.Url, KustoDataType.String),
+                    new KustoValue("TaskId", b.Raw.Task?.Id, KustoDataType.Int),
+                    new KustoValue("TaskName", b.Raw.Task?.Name, KustoDataType.String),
+                    new KustoValue("TaskVersion", b.Raw.Task?.Version, KustoDataType.String),
+                    new KustoValue("Attempt", b.Raw.Attempt, KustoDataType.Int),
+                });
+        }
+
+        public async Task WriteTimelineValidationMessages(IEnumerable<(int buildId, BuildRequestValidationResult validationResult)> validationResults)
+        {
+            await KustoHelpers.WriteDataToKustoInMemoryAsync(
+                ingest,
+                _database,
+                "TimelineIssues",
+                _logger,
+                validationResults,
+                b => new[]
+                {
+                    new KustoValue("BuildId", b.buildId, KustoDataType.Int),
+                    new KustoValue("RecordId", null, KustoDataType.String),
+                    new KustoValue("Index", null, KustoDataType.Int),
+                    new KustoValue("Path", null, KustoDataType.String),
+                    new KustoValue("Type", b.validationResult.Result, KustoDataType.String),
+                    new KustoValue("Category", "ValidationResult", KustoDataType.String),
+                    new KustoValue("Message", b.validationResult.Message, KustoDataType.String),
+                    new KustoValue("Bucket", "ValidationResult", KustoDataType.String),
+                });
+        }
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
If we "seal" the class, we don't need the complicated dispose pattern.  And the "disposedValue" isn't really necessary... if we get disposed twice, it's fine to just dispose the inner stuff twice too. :-)

Reply:
👍